test/Foo.purs

```diff
@@ -21,15 +22,13 @@ data Exp =
   | Lam String Exp
   | App Exp Exp
 
-derive instance genericExp :: Generic Exp
+derive instance genericExp :: Generic Exp _
 
-instance eqExp :: Eq Exp where
-  eq =
-    gEq
+derive instance eqExp :: Eq Exp
```

Comment:
We can derive `Eq` now. Dunno if we want to use the generic implementation instead. Happy to change it either way.